packages/menu-bar/src/vaadin-menu-bar-interactions-mixin.js

```diff
@@ -144,38 +145,75 @@ export const InteractionsMixin = (superClass) =>
 
     /**
      * @param {!KeyboardEvent} event
-     * @protected
+     * @private
      */
-    _onKeydown(event) {
+    _onArrowDown(event) {
+      // Prevent page scroll.
+      event.preventDefault();
+
       const button = this._getButtonFromEvent(event);
-      if (button) {
-        if (event.key === 'Escape') {
-          this._hideTooltip();
-        }
+      if (button === this._expandedButton) {
+        // Menu opened previously, focus first item
+        this._focusFirstItem();
+      } else {
+        this.__openSubMenu(button, event);
+      }
+    }
+
+    /**
+     * @param {!KeyboardEvent} event
+     * @private
+     */
+    _onArrowUp(event) {
+      // Prevent page scroll.
+      event.preventDefault();
+
+      const button = this._getButtonFromEvent(event);
+      if (button === this._expandedButton) {
+        // Menu opened previously, focus last item
+        this._focusLastItem();
+      } else {
+        this.__openSubMenu(button, event, { focusLast: true });
+      }
+    }
+
+    /**
+     * Override an event listener from `KeyboardMixin`:
+     * - to close the sub-menu for expanded button,
+     * - to close a tooltip for collapsed button.
+     *
+     * @param {!KeyboardEvent} event
+     * @protected
+     * @override
+     */
+    _onEscape(event) {
+      if (event.composedPath().includes(this._expandedButton)) {
+        this._close(true);
+      } else {
+        this._hideTooltip();
```

Comment:
Is it on purpose that `_hideTooltip()` is now in the `else` branch? Because it didn't previously.

Reply:
It seems its not needed when a menu is opened, as the tooltip is not visible in that case. The behavior is still the same as on `master` at least.

Reply:
But would it work the same in the case the tooltip was opened programmatically? Or is it not supported when using tooltip inside a menu-bar?

Reply:
The internal logic hides the tooltip when opening a menu, so that case is probably not intended / supported - the menu and the tooltip would probably overlay each other.
Even if you open it programmatically when the menu is already open, with the new behavior the first keypress would close the menu, the second would close the tooltip, so not a big issue probably.

Reply:
Moved the `_hideTooltip()` call out of the `else` branch to keep the code the same as previously.